docker-compose.yml

```diff
@@ -0,0 +1,25 @@
+version: '3.4'
+services:
+  weaviate:
+    container_name: "weaviate"
+    restart: always
+    image: semitechnologies/weaviate:1.28.2
+    ports:
+      - "8080:8080"
+      - "0.0.0.0:50051:50051"
+    environment:
+      QUERY_DEFAULTS_LIMIT: 25
+      AUTHENTICATION_ANONYMOUS_ACCESS_ENABLED: 'true'
+      PERSISTENCE_DATA_PATH: '/var/lib/weaviate'
+      DEFAULT_VECTORIZER_MODULE: 'none'
+      ENABLE_MODULES: ''
+      CLUSTER_HOSTNAME: 'node1'
+      LOG_LEVEL: 'debug'
+    volumes:
+      - weaviate_data:/var/lib/weaviate
+    extra_hosts:
+      - "host.docker.internal:host-gateway"
+
+volumes:
+  weaviate_data:
+    name: "weaviate_data"
```

Comment:
Why the addition of this file?
- We already have docker compose files defined in the `ci/` directory that are used when the tests run as GitHub action so this is duplicated
- This file is hard-coded to version `1.28.2`. How can we test that the changes work for other versions?
- The `extra_hosts` seems specific to MacOS distributions when the CI runs using Ubuntu so would be incompatible there